mods/d2k/maps/harkonnen-09b/harkonnen09b.lua

```diff
@@ -0,0 +1,412 @@
+	{ "trooper", "trooper", "trooper", "trooper", "trooper", "combat_tank_a" },
+	{ "combat_tank_a", "combat_tank_a", "quad", "quad", "trike" }
+}
+
+InitialCorrinoReinforcements =
+{
+	{ "trooper", "trooper", "trooper", "trooper", "quad", "quad" },
+	{ "trooper", "trooper", "trooper", "trooper", "trooper", "combat_tank_h", "combat_tank_h" },
+	{ "trooper", "trooper", "trooper", "combat_tank_h", "combat_tank_h", "combat_tank_h" }
+}
+
+AtreidesPaths =
+{
+	{ AtreidesEntry1.Location, AtreidesRally1.Location },
+	{ AtreidesEntry2.Location, AtreidesRally2.Location },
+	{ AtreidesEntry3.Location, AtreidesRally3.Location },
+	{ AtreidesEntry4.Location, AtreidesRally4.Location },
+	{ AtreidesEntry5.Location, AtreidesRally5.Location }
+}
+
+InitialAtreidesPaths =
+{
+	{ AConYard1.Location, AtreidesRally6.Location },
+	{ ARefinery2.Location, AtreidesRally7.Location },
+	{ AtreidesEntry6.Location, AtreidesRally8.Location }
+}
+
+InitialCorrinoPaths =
+{
+	{ CConYard.Location, CorrinoRally1.Location },
+	{ CHeavyFactory.Location, CorrinoRally2.Location },
+	{ CRefinery2.Location, CorrinoRally3.Location }
+}
+
+HarkonnenReinforcements = { "combat_tank_h", "combat_tank_h" }
+
+HarkonnenPath = { HarkonnenEntry.Location, HarkonnenRally.Location }
+
+SendStarportReinforcements = function()
+	Trigger.AfterDelay(CorrinoStarportDelay[Difficulty], function()
+		if CStarport.IsDead or CStarport.Owner ~= corrino_small then
+			return
+		end
+		
+		reinforcements = Utils.Random(CorrinoStarportReinforcements[Difficulty])
+
+		local units = Reinforcements.ReinforceWithTransport(corrino_small, "frigate", reinforcements, { CorrinoStarportEntry.Location, CStarport.Location + CVec.New(1, 1) }, { CorrinoStarportExit.Location })[2]
+		Utils.Do(units, function(unit)
+			unit.AttackMove(AtreidesAttackLocation)
+			IdleHunt(unit)
+		end)
+
+		SendStarportReinforcements()
+	end)
+end
+
+SendHarkonnenReinforcements = function(delay)
+	Trigger.AfterDelay(delay, function()
+		Reinforcements.ReinforceWithTransport(player, "carryall.reinforce", HarkonnenReinforcements, HarkonnenPath, { HarkonnenPath[1] })
+		Trigger.AfterDelay(DateTime.Seconds(5), function()
+			Media.PlaySpeechNotification(player, "Reinforce")
+		end)
+	end)
+end
+
+SendAirStrike = function()
+	if AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main then
+		return
+	end
+
+	local targets = Utils.Where(player.GetActors(), function(actor)
+		return
+			actor.HasProperty("Sell") and
+			actor.Type ~= "wall" and
+			actor.Type ~= "medium_gun_turret" and
+			actor.Type ~= "large_gun_turret" and
+			actor.Type ~= "wind_trap"
+	end)
+
+	if #targets > 0 then
+		AHiTechFactory.SendAirstrike(Utils.Random(targets).CenterPosition, true, 0)
+	end
+
+	Trigger.AfterDelay(DateTime.Minutes(5), SendAirStrike)
+end
+
+BuildFremen = function()
+	if APalace.IsDead or APalace.Owner ~= atreides_main then
+		return
+	end
+
+	APalace.Produce("fremen")
+	APalace.Produce("fremen")
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
```

Comment:
A delay of 1s is too short, the Fremen won't be "idle" and therefore won't be counted. Increasing it to 2s solved it for me. I don't know on what that behavior depends, so maybe it should be increased further...

Reply:
`Harkonnen 9a` would also benefit from this change.

Reply:
I think it would be better if we don't take much risk and make it 5 sec.